src/Monolog/Logger.php

```diff
@@ -311,6 +308,11 @@ public function addRecord(int $level, string $message, array $context = []): boo
             $record = call_user_func($processor, $record);
         }
 
+        reset($this->handlers);
+        while ($key !== key($this->handlers)) {
+            next($this->handlers);
```

Comment:
@staabm I do "something" - I loop until we reach the point where the `foreach` above completed its execution. Effectively - we are looking for the last processed key.